static/app/components/searchSyntax/parser.tsx

```diff
@@ -1405,6 +1407,12 @@ export type SearchConfig = {
    * Text filter keys we allow to have operators
    */
   textOperatorKeys: Set<string>;
+  /**
+   * A collection of secondary aliases for filter keys.
+   * These are used to ensure that the filter key does not show them as invalid, however
+   * they will not be shown in the filter key dropdown.
+   */
+  filterKeysSecondaryAliases?: TagCollection;
```

Comment:
I don't think you need to change anything in this file, you can just mash the secondary aliases into `supportedTags` along with the others. The syntax parser doesn't have anything to do with displaying filter options